pylabrobot/__init__.py

```diff
@@ -1,15 +1,61 @@
 import datetime
 import logging
+from pathlib import Path
+from typing import Union
 
 from pylabrobot.__version__ import __version__
 
-# Create a logger
-logger = logging.getLogger("pylabrobot")
-logger.setLevel(logging.DEBUG)
+from pylabrobot.config import load_config, Config
 
-# Add a file handler
-now = datetime.datetime.now().strftime("%Y%m%d")
-fh = logging.FileHandler(f"pylabrobot-{now}.log")
-fh.setLevel(logging.DEBUG)
-fh.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
-logger.addHandler(fh)
+CONFIG_FILE_NAME = "pylabrobot"
+
+CONFIG = load_config(CONFIG_FILE_NAME, create_default=True)
+"""The loaded configuration for pylabrobot."""
+
+
+def setup_logger(log_dir: Union[Path, str], level: int):
+  """
+  Set up the logger for pylabrobot. If the log_dir does not exist, it will be created.
+
+  Args:
+    log_dir: The directory to store the log files.
+    level: The logging level.
+
+  """
+  # Create a logger
+  if isinstance(log_dir, str):
+    log_dir = Path(log_dir)
+  if not log_dir.exists():
+    log_dir.mkdir(parents=True)
+  logger = logging.getLogger("pylabrobot")
+  logger.setLevel(level)
+
+  now = datetime.datetime.now().strftime("%Y%m%d")
+  # remove file handler if it exists
+  if len(logger.handlers) > 0:
+    logger.handlers.clear()
```

Comment:
what is the rationale for this?

Reply:
This lets `pylabrobot.configure` be called after importing to update the project's logger. It comes into play with the testing fixture I put in for sequestering all test logs into test_logs, if I make it:

```python
@pytest.fixture(autouse=True)
def setup_test_config():
  logger = logging.getLogger("pylabrobot")
  print("initial logging handlers", logger.handlers[0])
  configure(TEST_CONFIG)
  print("testing logging handlers", logger.handlers[0])
  yield

```
I get

```
 initial logging handlers <FileHandler ~/PycharmProjects/pylabrobot/pylabrobot/config/pylabrobot-20240605.log (INFO)>
testing logging handlers <FileHandler ~/PycharmProjects/pylabrobot/pylabrobot/config/test_logs/pylabrobot-20240605.log (INFO)>
```

without this check, it would end up with both loggers being used.

Reply:
makes sense, thanks for explaining :)